src/coreComponents/constitutive/fluid/PVTFunctions/PVTFunctionHelpers.hpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file PVTFunctionHelpers.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_FLUID_PVTFUNCTIONS_PVTFUNCTIONHELPERS_HPP
+#define GEOSX_CONSTITUTIVE_FLUID_PVTFUNCTIONS_PVTFUNCTIONHELPERS_HPP
+
+namespace geosx
+{
+
+namespace constitutive
+{
+
+namespace PVTProps
+{
+
+class PTTableCoordinates
+{
+public:
+  PTTableCoordinates()
+  { coords.resize( 2 ); }
+
+  localIndex nPressures() const { return coords[0].size(); }
+  localIndex nTemperatures() const { return coords[1].size(); }
+
+  void appendPressure( const real64 & pres ) { coords[0].emplace_back( pres ); }
+  void appendTemperature( const real64 & temp ) { coords[1].emplace_back( temp ); }
+
+  real64 const & getPressure( localIndex i ) const { return coords[0][i]; }
+  real64 const & getTemperature( localIndex i ) const { return coords[1][i]; }
+
+  array1d< array1d< real64 > > const & getCoords() const { return coords; }
+
+private:
+  array1d< array1d< real64 > > coords;
+};
```

Comment:
So do you think it is worth the shot or not?
I do not code the algorithms so I cannot be sure it's useful. You tell me.

Another solution could be to inherit
```c++
class PTTableCoordinates: public array1d< array1d< real64 > >
```
so you could use `PTTableCoordinates` for what it is (an array of array) too.
That would give more possibilities so this may (not?) be what you want.
You shouldn't need the `getCoords` anymore.
What would you prefer?
Also, we should do it `geosx` style, which I don't know.

Last, you should possibly define some `0` and `1` private constants.

Reply:
Here I think the two ways that you proposed (either inheriting from `array1d< array1d< real64 > >`  or have `array1d < array1d< real64 > >` as datamember) are quite similar for what we want to do: avoiding hard-coded 0 and 1 everywhere in the `PVTFunction`s. I just implemented the simplest option for me, which is to have the `array1d< array1< real64 > >` as datamember. To be honest, I think it is fine like that for now.

Also, I added the private constants that you suggested in the more recent commit.